extension/src/util/config.ts

```diff
@@ -26,6 +35,72 @@ export function getConfigJavaImportGradleJavaHome(): string | null {
     return vscode.workspace.getConfiguration("java").get<string | null>("import.gradle.java.home", null);
 }
 
+export async function getJavaExecutablePath(): Promise<string | undefined> {
+    const REQUIRED_JDK_VERSION = 17;
+
+    // search from jdt.ls.java.home, java.home
+    const javaHomeGetters = [getJdtlsConfigJavaHome, getConfigJavaHome];
+    let javaHome: string | undefined = undefined;
+    let javaVersion = 0;
+
+    for (const getJavaHome of javaHomeGetters) {
+        javaHome = getJavaHome() || undefined;
+        if (javaHome) {
+            javaVersion = await getMajorVersion(javaHome);
+            // Ensure the Java version is greater than 17
```

Comment:
No need to add comment here. Because it's very clear that you are ensuring the java version is higher than 17. Besides, imagine that in the future, if the required version changes, you will need to update this comment as well.

I would suggest to adding comment at line 39, explaining where does the requirement comes from